src/App.js

```diff
@@ -29,37 +30,103 @@ const App = () => {
   // This starts state off as a 2D array of JS objects with
   // empty value and unique ids.
   const [squares, setSquares] = useState(generateSquares());
+  const [currentPlayer, setCurrentPlayer] = useState(PLAYER_1);
+  // const [winner, setWinner] = useState(null);
+  // const [gameEnd, setGameEnd] = useState(false);
 
-  // Wave 2
+  const squareClicked =(id)=>{
+    let didPlay = false 
+    if (winner !== null) {return}
+    const newState = squares.map(row =>row.map(pos =>{
```

Comment:
👍 
We map both the outer and inner list. If we end up making a play, we don't want to modify the existing state directly. We want to make a new state, and notify React about it through setSquares.